hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/DistributedScheduledExecutorService.java

```diff
@@ -359,4 +375,60 @@ private void sendBatch(int partitionId, String name, List<ScheduledExecutorMerge
             invoke(SERVICE_NAME, operation, partitionId);
         }
     }
+
+    private class StatsLoggerThread extends HazelcastManagedThread {
+        private final Map<String, MutableInteger> tasksSizes = new HashMap<String, MutableInteger>();
+        private final ILogger logger = nodeEngine.getLogger(DistributedScheduledExecutorService.class);
+        private final IPartitionService partitionService = nodeEngine.getPartitionService();
+        private final long logDelayMillis;
+        private volatile boolean shutdown;
+
+        public StatsLoggerThread(int logDelaySeconds) {
+            this.logDelayMillis = TimeUnit.SECONDS.toMillis(logDelaySeconds);
+        }
+
+
+        private void shutdown() {
+            shutdown = true;
+            interrupt();
+        }
+
+        @Override
+        public void run() {
+            while (!shutdown) {
+                try {
+                    Thread.sleep(logDelayMillis);
+                } catch (InterruptedException e) {
+                    continue;
+                }
+
+                tasksSizes.clear();
+                for (ScheduledExecutorPartition partition : partitions) {
+                    if (partition == null) {
+                        continue;
+                    }
+
+                    if (partitionService.getPartitionOwner(partition.partitionId()) != nodeEngine.getThisAddress()) {
```

Comment:
This can be replaced with `partition.isLocal()`. Also, it might deserve a note somewhere that the reported sizes might be inaccurate during a migration.